SPECS/python-async-generator/python-async-generator.spec

```diff
@@ -0,0 +1,103 @@
+# what it's called on pypi
+%global srcname async_generator
+# what it's imported as
+%global libname async_generator
+# name of egg info directory
+%global eggname async_generator
+# package name fragment
+%global pkgname async-generator
+
+%global _description \
+This library generally tries hard to match the semantics of Python 3.6's native\
+async generators in every detail (PEP 525), with additional support for yield\
+from and for returning non-None values from an async generator (under the\
+theory that these may well be added to native async generators one day).
+
+Summary:        Async generators and context managers
+Name:           python-%{pkgname}
+Version:        1.10
+Release:        10%{?dist}
+License:        MIT or ASL 2.0
+URL:            https://github.com/python-trio/async_generator
+Vendor:         Microsoft
+Distribution:   Mariner
+Source0:        %{url}/archive/v%{version}/%{srcname}-%{version}.tar.gz
+BuildArch:      noarch
+
+
+%description %{_description}
+
+
+%package -n python%{python3_pkgversion}-%{pkgname}
+Summary:        %{summary}
+BuildRequires:  python%{python3_pkgversion}-devel
+BuildRequires:  python%{python3_pkgversion}-setuptools
+%if %{with_check}
+BuildRequires:  python3-pip
+BuildRequires:  python3-atomicwrites
+%endif
+%{?python_provide:%python_provide python%{python3_pkgversion}-%{pkgname}}
+
+
+%description -n python%{python3_pkgversion}-%{pkgname} %{_description}
+
+
+%prep
+%autosetup -n %{srcname}-%{version}
+rm -r %{eggname}.egg-info
+
+
+%build
+%py3_build
+
+
+%install
+%py3_install
+
+
+%check
+pip3 install atomicwrites>=1.3.0 \
+    attrs>=19.1.0 \
+    more-itertools>=7.0.0 \
+    pluggy>=0.11.0 \
+    pytest>=5.4.0 \
+    pytest-cov>=2.7.1 
```

Comment:
Is there a "test-requirements.txt" (or similar) file in the source that we could pull these from, instead of having them hardcoded?

Reply:
There is, but it doesn't include the dependencies for pytest or the versions